### PR TITLE
Publish operator binaries from release script

### PR DESCRIPTION
**Pull request by lgarciaaco**
With this PR, when doing a release by running `syndesis release` (not a snapshot release), a [release in github](https://github.com/syndesisio/syndesis/releases) is created and the operator binaries are included.

---
How to use:
```
 syndesis release --release-version tag_name 
```
Replace tag_name with with the real tag, 1.9 ... for instance

---
What produces:
The final releases will be similar to like [this](https://github.com/lgarciaaco/syndesis/releases). The body of the release is generated with [gren](https://github.com/github-tools/github-release-notes), further tuning of parameters can be done in the future. 

The release will be created when `syndesis release` is ran without `--snapshot-release`

### CODE REVIEW

**Comment by lgarciaaco** [comment]
Original PR: https://github.com/syndesisio/syndesis/pull/7012

File: tools/bin/commands/release [hunk 1]
```diff
@@ -235,6 +240,88 @@ update_image_versions() {
...
+    check_github_access_token
+    check_github_username
+    # check_gren_access_token
+
+    local data="{\
+        \"tag_name\": \"${tag}\", \
+        \"name\": \"${release_name}\" \
+    }"
+
+	local upload_url=$(curl -q -s --fail \
```
Comment (zregvart) [inline tools/bin/commands/release]: tab

File: tools/bin/commands/release [hunk 2]
```diff
@@ -235,6 +240,88 @@ update_image_versions() {
...
+    local release_dir=$1
+    local tag=$2
+
+    check_github_access_token
+    check_github_username
+    # check_gren_access_token
+
+    local data="{\
+        \"tag_name\": \"${tag}\", \
+        \"name\": \"${release_name}\" \
```
Comment (zregvart) [inline tools/bin/commands/release]: Not sure $release_name is defined anywhere, perhaps

```suggestion
        \"name\": \"${tag}\" \
```

File: tools/bin/commands/release [hunk 3]
```diff
@@ -235,6 +240,88 @@ update_image_versions() {
...
+
+publish_artifacts() {
+    local release_dir=$1
+    local tag=$2
+
+    check_github_access_token
+    check_github_username
+    # check_gren_access_token
+
+    local data="{\
```
Comment (zregvart) [inline tools/bin/commands/release]: Would be good to set the `prerelease` to true for snapshot releases.

**Comment by lgarciaaco** [comment]
Fixes https://github.com/syndesisio/syndesis/issues/6976

**Pull request merged**